lib/elixir/lib/calendar/date.ex

```diff
@@ -398,6 +398,76 @@ defmodule Date do
     end
   end
 
+  @doc """
+  Converts a number of gregorian days to a `Date` struct.
+
+  ## Examples
+
+      iex> Date.from_gregorian_days(1)
+      {:ok, ~D[0000-01-02]}
+      iex> Date.from_gregorian_days(730_485)
+      {:ok, ~D[2000-01-01]}
+      iex> Date.from_gregorian_days(-1)
+      {:error, :invalid_days}
+
+  """
+  @spec from_gregorian_days(non_neg_integer(), Calendar.calendar()) :: {:ok, t} | {:error, atom}
+  def from_gregorian_days(days, calendar \\ Calendar.ISO)
+
+  def from_gregorian_days(days, calendar) when is_integer(days) and days >= 0 do
```

Comment:
Curious why you disallow negative days given that dates extend backwards as well and are supported by the calendar behaviour?

Reply:
Correct, we should allow negative days. :)